writer.go

```diff
@@ -1439,7 +1452,19 @@ func (c *ColumnWriter) newColumnBuffer() ColumnBuffer {
 	case c.maxRepetitionLevel > 0:
 		column = newRepeatedColumnBuffer(column, c.maxRepetitionLevel, c.maxDefinitionLevel, nullsGoLast)
 	case c.maxDefinitionLevel > 0:
-		column = newOptionalColumnBuffer(column, c.maxDefinitionLevel, nullsGoLast)
+		// Since these buffers are pooled, we can afford to allocate a bit more memory in
+		// order to reduce the risk of needing to resize the buffer.
+		var (
+			size             = int(float64(column.Cap()) * 1.5)
+			rows             = int32Buffers.get(size)
+			definitionLevels = buffers.get(size)
+		)
+		c.closers = append(c.closers, closerFunc(func() error {
```

Comment:
Is there a guarantee that the closer will be called at most once?

If the code can call it more than once it would cause an invalid de reference.